server/graphql/stocktake_line/src/stocktake_line_queries.rs

```diff
@@ -0,0 +1,337 @@
+    /// ascending)
+    pub desc: Option<bool>,
+}
+
+impl StocktakeLineSortInput {
+    pub fn to_domain(self) -> StocktakeLineSort {
+        use StocktakeLineSortField as to;
+        use StocktakeLineSortFieldInput as from;
+        let key = match self.key {
+            from::ItemCode => to::ItemCode,
+            from::ItemName => to::ItemName,
+            from::Batch => to::Batch,
+            from::ExpiryDate => to::ExpiryDate,
+            from::PackSize => to::PackSize,
+            from::LocationName => to::LocationName,
+        };
+
+        StocktakeLineSort {
+            key,
+            desc: self.desc,
+        }
+    }
+}
+
+#[derive(Union)]
+pub enum StocktakesLinesResponse {
+    Response(StocktakeLineConnector),
+}
+
+pub fn stocktake_lines(
+    ctx: &Context<'_>,
+    store_id: &str,
+    page: Option<PaginationInput>,
+    filter: Option<StocktakeLineFilterInput>,
+    sort: Option<StocktakeLineSortInput>,
+) -> Result<StocktakesLinesResponse> {
+    let user = validate_auth(
+        ctx,
+        &ResourceAccessRequest {
+            resource: Resource::QueryStocktake,
+            store_id: Some(store_id.to_string()),
+        },
+    )?;
+
+    let service_provider = ctx.service_provider();
+    let service_ctx = service_provider.context(store_id.to_string(), user.user_id)?;
+    let service = &service_provider.stocktake_line_service;
+
+    let stocktake_filter = filter.clone().map(StocktakeLineFilter::from);
+
+    let mut stocktake_id = "".to_string();
+    if let Some(f) = stocktake_filter {
+        if let Some(id) = f.stocktake_id {
+            if let Some(id) = id.equal_to {
+                stocktake_id = id;
+            }
+        }
+    }
+
+    let stocktake_lines = service.get_stocktake_lines(
+        &service_ctx,
+        store_id,
+        &stocktake_id,
+        page.map(PaginationOption::from),
+        filter.map(StocktakeLineFilter::from),
+        sort.map(|s| s.to_domain()),
+    );
+
+    if let Ok(stocktake_lines) = stocktake_lines {
+        Ok(StocktakesLinesResponse::Response(StocktakeLineConnector {
+            total_count: stocktake_lines.count,
+            nodes: stocktake_lines
+                .rows
+                .into_iter()
+                .map(|line| StocktakeLineNode::from_domain(line))
+                .collect(),
+        }))
+    } else {
+        let err = stocktake_lines.unwrap_err();
+        let formatted_error = format!("{:#?}", err);
+        let graphql_error = match err {
+            GetStocktakeLinesError::DatabaseError(err) => err.into(),
+            GetStocktakeLinesError::InvalidStore => {
+                StandardGraphqlError::BadUserInput(formatted_error)
+            }
+            GetStocktakeLinesError::InvalidStocktake => {
+                StandardGraphqlError::BadUserInput(formatted_error)
+            }
+            GetStocktakeLinesError::ListError(err) => match err {
```

Comment:
I think just `return Err(list_error_to_gql_error(err))` ?

Reply:
I had used that method before but was getting incompatible error types. It expects the `StandardGraphqlError` error type to be returned. But the `` method returns the `async_graphql::Error` error type.

Reply:
05e96188b